### DISCUSSION TITLE
Two submit buttons with different validation?

### DISCUSSION THREAD

**Discussion by Falci** [react-hook-form/react-hook-form#6349] [Q&A]
Hi,

I'm using RHF with Yup. I have a form with some string properties:

```
type Form = {
  foo: string;
  bar?: string;
}
```

Also, this form has two CTA: `Preview` and `Save`.
For **preview**, `bar` is optional but it should be required for **save**

How can I handle it?

**Comment by bluebill1049** [comment]
if you have using schema validation `resovler` then you can switch the schema with `context`.
